### PR TITLE
Maintenance

### PR DESCRIPTION
**Pull request by stroblme**
This PR introduced proper handling for GPU and CPU installations.
It adds documentation how to handle those scenarios with poetry and adds a separate requirements.in file for GPU systems.

### CODE REVIEW

**Review by eileen-kuehn — COMMENTED** [review]
Great stuff, I left some comments. So please have a look before merging.

**Comment by stroblme** [comment]
Additional information: The PR actually does not fully solve the issue with Poetry and PyTorch. Original idea was to have a keyword "gpu" that can be provided as extra parameter to allow switching between CPU and GPU installation.
However, there is an issue with the current version of poetry [here](https://github.com/python-poetry/poetry/issues/7748) which prevents this approach from working.
As I expect this to be fixed in the near future, I just disabled the line with the GPU source (see review comments by @eileen-kuehn ) to simply un-comment it when Poetry fixes aforementioned issue.

File: pyproject.toml [hunk 1]
```diff
@@ -5,36 +5,42 @@ description = ""
...
+
 [tool.poetry.dependencies]
 python = ">=3.9,<3.11"
 pennylane = "^0.31.0"
 pillow = "^10.0.0"
 scikit-image = "^0.21.0"
 ipykernel = "^6.25.0"
 matplotlib = "^3.7.2"
+torch = [
+#    {version = "2.0.1", source = "torch_gpu", markers = "extra == 'gpu'"},
```
[the remark this answers is not in the document]
Comment (stroblme) [reply pyproject.toml]: Thanks for the review and the hint! I was missing some documentation here, see my comment below.

File: pyproject.toml [hunk 2]
```diff
@@ -5,36 +5,42 @@ description = ""
...
+    {version = "2.0.1+cpu", source = "torch_cpu", markers = "extra != 'gpu'"}
+]
 torchvision = [
-    {version = "^0.15.2+cpu", source = "torch_cpu", markers = "extra == 'cpu'"},
-    {version = "^0.15.2", source = "torch_gpu", markers = "extra == 'gpu'"}
+#    {version = "0.15.2", source = "torch_gpu", markers = "extra == 'gpu'"},
```
[the remark this answers is not in the document]
Comment (stroblme) [reply pyproject.toml]: See above..

**Comment by cDenius** [comment]
Maybe add an issue to fix the poetry issue? Just so it is not forgotten until they fixed it.

**Pull request merged**